src/types/account/index.ts

```diff
@@ -96,6 +97,9 @@ export type AccountSettingsAttrs = {
 	[key: string]: string | number | Array<string | number> | undefined;
 };
 
+export type UpdateSettings = (settingsMods: Partial<Mods>) => void;
```

Comment:
This is still using the Mods type, wouldn't it be better to use the AccountSettings? The Mods type includes additional fields which are not updated in the implementation you made. The AccountSettings is instead the exact representation of what will be updated.
Another solution could be creating a specific type for this function arg

Reply:
> I'll add an info. We should try to start using interfaces to declare types that could be extended (and remove the generic string key that is present right now only for compatibility with the previous types), so that modules can augment them to add fields strictly related to the module itself. Specifically, if you need to extend the AccountSettingsPrefs, or the AccountSettingsAttrs, you can
> 
> 1. refactor the type to be an interface if it is not already
> 2. export it if not already exported
> 3. augment the interface inside the module, in order to add the specific fields for that interface
> 
> A practical example Let's say you are managing an account pref not listed inside the AccountSettingsPrefs of the shell. You can have it type-checked by creating the module declaration for the shell, and extend in it the interface
> 
> ```ts
> import * as Shell from '@zextras/carbonio-shell-ui';
> 
> declare module '@zextras/carbonio-shell-ui' {
> 	interface AccountSettingsPrefs extends Shell.AccountSettingsPrefs {
> 		yourModulePref: 'value1' | 'value2';
> 	}
> }
> ```

We can talk about it but for now it's out of context considering the current task

Reply:
> This is still using the Mods type, wouldn't it be better to use the AccountSettings? The Mods type includes additional fields which are not updated in the implementation you made. The AccountSettings is instead the exact representation of what will be updated. Another solution could be creating a specific type for this function arg

We know that the current type is not optimal, but changing it will have a bigger impact on the existing code.
Our approach has been a conservative one, re-using the existing types, to not change too much untested code with the risk of producing BCs.